aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -83,21 +84,21 @@ public boolean hasPendingMutation(@NonNull String modelId) {
         String modelId = incomingMutation.getMutatedItem().getId();
         @SuppressWarnings("unchecked")
         PendingMutation<T> existingMutation = (PendingMutation<T>) nextMutationForModelId(modelId);
-        if (existingMutation == null || inFlightMutations.contains(existingMutation.getMutationId())) {
-            return save(incomingMutation).andThen(notifyContentAvailable());
-        }
-        switch (incomingMutation.getMutationType()) {
-            case CREATE:
-                return handleCreation();
-            case UPDATE:
-                return handleUpdate(existingMutation, incomingMutation);
-            case DELETE:
-                return handleDeletion(existingMutation, incomingMutation);
-            default:
-                return unknownMutationType(incomingMutation.getMutationType());
+        if (existingMutation == null) {
```

Comment:
Did you mean to remove the inFlightMutations condition? I thought that was the piece Jameson just added.

Reply:
I had moved that check down to one of the methods that handles conflicts. However, I think you're correct. It probably makes more sense here.